source/documentation/deploying_services/sqs.md

```diff
@@ -0,0 +1,342 @@
+
+You can bind an SQS queue using the command line or the manifest file.
+
+Run the following to bind an SQS queue to your app:
+
+```
+cf bind-service APP_NAME SERVICE_NAME
+```
+
+It will take between 1 and 5 minutes to set up the binding. To check its progress, run:
+
+```
+cf service SERVICE_NAME
+```
+
+where:
+
+- `APP_NAME` is the name of a deployed instance of your app, as specified in your app's manifest or push command
+- `SERVICE_NAME` is a unique descriptive name for this SQS queue
+
+When binding an SQS service to your app, you have the option of restricting the
+queue permissions given to that app. You can control this using `bind-service`'s `-c`
+flag to set the bind parameters `access_policy` JSON key, for example:
+
+```
+cf bind-service APP_NAME SERVICE_NAME -c '{"access_policy": "consumer"}'
+```
+
+The currently supported `access_policy` values are:
+
+- `full`, the default, full access permissions
+- `producer` is appropriate for an app which will be sending messages to the queue,
+  but does not need to be able to read or delete messages.
+- `consumer` is appropriate for apps which will be reading and possibly deleting messages,
+  but do not need to be able to send messages to the queue.
+
+You can use the [app's manifest](https://docs.cloud.service.gov.uk/deploying_apps.html#deploying-public-apps)
+to bind an SQS queue to the app (with default `full` permissions only).
+It will bind automatically when you next deploy your app. An example manifest:
+
+```
+--
+applications:
+- name: APP_NAME
+  services:
+  - SERVICE_NAME
+```
+
+### Connect to an SQS queue from your app
+
+Restart your app to populate your `VCAP_SERVICES` [environment variable](/deploying_apps.html#system-provided-environment-variables)
+with the credentials needed to connect to the SQS queue.
+
+An example `VCAP_SERVICES` environment variable:
+
+```
+{
+ "VCAP_SERVICES": {
+  "aws-sqs-queue": [
+   {
+    "binding_name": null,
+    "credentials": {
+     "aws_access_key_id": "XXXXXXXXXXXXXXXXXXXX",
+     "aws_region": "eu-west-2",
+     "aws_secret_access_key": "XXXX/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+     "primary_queue_url": "https://XXXXXXXXXXXXX",
+     "secondary_queue_url": "https://XXXXXXXXXXXXX"
+    },
+    "instance_name": "my-queue",
+    "label": "aws-sqs-queue",
+    "name": "my-queue",
+    "plan": "standard",
+    "provider": null,
+    "syslog_drain_url": null,
+    "tags": [
+     "sqs"
+    ],
+    "volume_mounts": []
+   }
+  ]
+ }
+}
+```
+
+You must pass credentials from your `VCAP_SERVICES` environment variable to
+your AWS software development kit (SDK). Use the appropriate [Amazon SDK
+documentation](https://aws.amazon.com/developer/tools/#sdks), and extract the
+credentials from VCAP_SERVICES to provide to the SDK. You can usually do this
+either by setting environment variables such as AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY or AWS_REGION, or by providing the SDK with the right static credentials
```

Comment:
Should these be in code blocks like VCAP_SERVICES is?

Reply:
yes they should...done